resources/data-provider.mjs

```diff
@@ -0,0 +1,151 @@
+// example url for local testing:
+// http://localhost:8080/?config=http://localhost:8080/resources/config.json
+import { defaultSuites } from "./tests.mjs";
+import { params } from "./shared/params.mjs";
+
+const DEFAULT_TAGS = ["all", "default", "experimental"];
+const DISALLOWED_DOMAINS = ["browserbench.org", "www.browserbench.org"];
+export class DataProvider {
+    _tags = new Set(DEFAULT_TAGS);
+    _suites = [];
+
+    get tags() {
+        return this._tags;
+    }
+
+    get suites() {
+        return this._suites;
+    }
+
+    /**
+     * Checks if a given string is a valid URL, supporting both absolute and relative paths.
+     *
+     * This function attempts to construct a URL object. For relative paths, it uses
+     * a dummy base URL to allow the URL constructor to parse them successfully.
+     *
+     * @param {string} url The URL string to validate.
+     * @returns {boolean} True if the URL is valid (absolute or relative), false otherwise.
+     */
+    _isValidUrl(url) {
+        if (typeof url !== "string" || url.length === 0)
+            return false;
+
+        try {
+            new URL(url, "http://www.example.com");
+            return true;
+        } catch (error) {
+            return false;
+        }
+    }
+
+    _freezeSuites() {
+        Object.freeze(this._suites);
+        this._suites.forEach((suite) => {
+            if (!suite.tags)
+                suite.tags = [];
+            if (suite.url.startsWith("experimental/"))
+                suite.tags.unshift("all", "experimental");
+            else
+                suite.tags.unshift("all");
+            suite.enabled = suite.tags.includes("default");
+            Object.freeze(suite.tags);
+            Object.freeze(suite.steps);
+        });
+    }
+
+    _freezeTags() {
+        Object.freeze(this._tags);
+    }
+
+    async init() {
+        if (params.config) {
+            try {
+                const benchmarkUrl = new URL(window.location);
+                // Don't fetch if the URL is from DISALLOWED_DOMAINS
+                if (DISALLOWED_DOMAINS.includes(benchmarkUrl.hostname)) {
```

Comment:
Please disallow all subdomains of `browserbench.org` at minimum.
Something like `benchmarkUrl.hostname.endsWith('browserbench.org')`.